raft/src/main/java/org/apache/kafka/raft/QuorumConfig.java

```diff
@@ -99,6 +100,11 @@ public class QuorumConfig {
     public static final String QUORUM_RETRY_BACKOFF_MS_DOC = CommonClientConfigs.RETRY_BACKOFF_MS_DOC;
     public static final int DEFAULT_QUORUM_RETRY_BACKOFF_MS = 20;
 
+    public static final String QUORUM_AUTO_JOIN_ENABLE = QUORUM_PREFIX + "auto.join.enable";
+    public static final String QUORUM_AUTO_JOIN_ENABLE_DOC = "Controls whether a KRaft controller should automatically " +
+        "join the cluster metadata partition for its cluster id.";
+    public static final boolean DEFAULT_QUORUM_AUTO_JOIN_ENABLE = false;
```

Comment:
`KafkaConfig` should validate that this `true` only when `process.role` contains `controller`.